js/app/modules/card/defaultFamily.js

```diff
@@ -192,21 +291,33 @@ const Thumb = new Module.Class({
     vfunc_size_allocate: function (alloc) {
         this._layout.clear_constraints();
```

Comment:
so I spent pretty much all day after presentation on this. I thought it would be straightforward but it's actually non-trivial.

>  1) create ALL constraints at _init (once) and then 2) enable/disable groups of constraints (the API allow this).

To do this you essentially have to simulate every possible configuration of the card at init and generate constraints for them, e.g. thumb card, horizontal alignment with synopsis showing; thumb card, horizontal alignment with synopsis not showing; thumb card, vertical alignment, context widget showing; etc etc.

It gets really gnarly trying to maintain all of those constraints and keep track of which one belongs to which situation. There's probably a clean-ish way to do it, but I wasn't able to find it.

Will try some more tomorrow